pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -446,7 +446,7 @@ func (ctrl *csiSnapshotSideCarController) updateSnapshotContentStatus(
 			SnapshotHandle: &snapshotHandle,
 			ReadyToUse:     &readyToUse,
 			CreationTime:   &createdAt,
-			RestoreSize:    &size,
+			RestoreSize:    size,
```

Comment:
I wonder if all the pointers to `size int64` is really needed. `size == 0` already indicates that the value is optional from a CSI Spec point of view:
```
// Information about a specific snapshot.
message Snapshot {
  // This is the complete size of the snapshot in bytes. The purpose of
  // this field is to give CO guidance on how much space is needed to
  // create a volume from this snapshot. The size of the volume MUST NOT
  // be less than the size of the source snapshot. This field is
  // OPTIONAL. If this field is not set, it indicates that this size is
  // unknown. The value of this field MUST NOT be negative and a size of
  // zero means it is unspecified.
  int64 size_bytes = 1;
```

This indicates an existing bug when `RestoreSize: 0` is set. It should probably never be set to 0 at all.

Reply:
that also works, remove the pointer and added `size >0` check